### ISSUE TITLE
Unable to build. librtmfp not found using pkg-config

### ISSUE THREAD

**Issue by KattaLord** [MonaSolutions/librtmfp#17]
When trying to build ffmpeg with librtmfp on a RaspberryPi 4 running Raspbian 10, configure gives me an error message telling me that librtmfp can not be found. 
Manually checking the version using pkg-config returns a version number higher than the required version.

I simply followed the installation instructions. Is there anything I am missing?

**Comment by thomasjammet** [comment]
Hi!

Did you install librtmfp with "sudo make install"?
The [FFmpeg fork](https://github.com/thomasjammet/FFmpeg/) supports all version superior to 2.14.1
Also can you tell me the result of the following command :
`whereis librtmfp`

Thanks

**Comment by KattaLord** [comment]
Hi.

I am quite sure that I did install it with `sudo make install`. Maybe without `sudo`, I am not entirely sure. The only thing I am certain is that there was no error in the console, if that helps. Could missing permissions cause this issue? 
The result for the `whereis` is as follows:
`librtmfp: /usr/lib/librtmfp.so /usr/include/librtmfp`
The result for `pkg-config --modversion librtmfp` is `2.15.3`, the results for `pkg-config --libs librtmfp` are
`-L/usr/lib/ -lrtmfp -lssl -lcrypto`

Hope that helps in resolving the issue.

**Comment by thomasjammet** [comment]
Ok thanks, it seems that librtmfp is correctly installed.
Can you send us the configure command you did and the result?

I have just tried on my raspberry pi with raspbian 9 and it worked directly.

**Comment by KattaLord** [comment]
I used the configure command as in the INSTALL.md: `./configure --disable-x86asm --enable-librtmp --enable-librtmfp --enable-libspeex --enable-libx264 --enable-libx265 --enable-debug --enable-gpl && make`.
The output said, that pkg-config can not find librtmfp >= 2.14.1.
I also tried removing various parameters until at some point, I only left `./configure --enable-librtmfp && make`.

Sadly, I can not try to build ffmpeg on that particular raspberry pi right now, as I switched to using SRT for that project now. 
I thought I would still create the Issue to see if it may be a bug in the software or just me doing something wrong. Or maybe something with the configuration of the raspberry. 
However, I have another raspberry pi 4 with the most recent raspbian lying around. I will try building ffmpeg with librtmfp on that tomorrow and tell you whether I was successful or not. If it works on that device, I would assume that it is an error with that particular pi's configuration or something like that.

**Comment by thomasjammet** [comment]
It seems to be an issue with pkg-config so.
Can you try the following command ?
`pkg-config --variable pc_path pkg-config | cut -d ':' -f 1`

And then print the content of librtmfp pkg-config file :
` cat "$(pkg-config --variable pc_path pkg-config | cut -d ':' -f 1)/librtmfp.pc"`

Thanks for your help, SRT is a great project too and sadly RTMFP suffers from the fact that Flash is gone from browsers.

**Comment by KattaLord** [comment]
The path is `/usr/local/lib/arm-linux-gnueabihf/pkgconfig` in case that helps.
The content of the file is:
```
prefix=/usr
exec_prefix=${prefix}
libdir=/usr/lib/
incdir=${prefix}/include

Name: librtmfp
Description: RTMFP client implementation
Version: 2.15.3
Requires: libssl,libcrypto
Libs: -L${libdir} -lrtmfp
Libs.private:
Cflags: -I${incdir}
```

Hope that helps. We initially wanted to switch from nginx and RTMP to MonaServer and RTMFP, as we had several problems using RTMP. For now we decided to use SRT instead as it was the fastest way to "fix" our problem.

**Comment by thomasjammet** [comment]
Thanks, it looks good, there should be no problem with the configure script. Maybe the ffmpeg `configure` script has been run in a different shell session where pkg-config was not up-to-date, otherwise I can't tell what's going wrong with your command.

What about your other raspberry pi 4?

**Comment by KattaLord** [comment]
Sorry for taking so long with it, I have been a bit busy and could try it just now.
The other pi gives me the same error: `ERROR: librtmfp >= 2.14.1 not found using pkg-config`.

**Comment by ROBERT-MCDOWELL** [comment]
do you have maybe several pkg-config on /usr/local/lib and /usr/lib64? several ffmpeg libs too?

**Comment by thomasjammet** [comment]
Yes it's a possibility, or maybe librtmfp is not correctly installed.
Can you compile TestClient which is in the librtmfp directory?
`cd TestClient && make`

Then try to run it :
`./TestClient`

And move TestClient to another directory (your home directory for example) and run ldd :
```
mv ./TestClient ~/
cd ~/ && ldd ./TestClient
```

Note the `librtmfps.so` location, it should be something like /usr/lib/librtmfp.so

Then try to run TestClient again :
`~/TestClient`
